### PR TITLE
Do not print new-build steps for alternative output formats

### LINKED ISSUE #6136: Output from `oc new-build -o json` should contain JSON only
The alternative output formats to `oc new-app` and `oc new-build` should not print the descriptive information to stdout. With the extra output, one cannot redirect the output to a file and use it later without having to manually edit it.

Example:

```
$ oc new-build -D 'FROM centos:7' -o yaml > build.yaml
$ head build.yaml 
--> Found Docker image 0e23418 (6 weeks old) from Docker Hub for "centos:7"
    * An image stream will be created as "centos:7" that will track this image
    * A Docker build using a predefined Dockerfile will be created
      * The resulting image will be pushed to image stream "centos:latest"
      * Every time "centos:7" changes a new build will be triggered
apiVersion: v1
items:
- apiVersion: v1
  kind: ImageStream
  metadata:
```

### PR DESCRIPTION
**Pull request by rhcarvalho**
Fixes #6136.

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@bparees @smarterclayton PTAL

[test]

File: pkg/cmd/cli/cmd/newapp.go [hunk 1]
```diff
@@ -215,39 +211,38 @@ func RunNewApplication(fullName string, f *clientcmd.Factory, out io.Writer, c *
...
 	var afterFn configcmd.AfterFunc
-	switch {
-	// only print success if we don't have installables
-	case !result.GeneratedJobs:
-		afterFn = configcmd.NewPrintNameOrErrorAfterIndent(mapper, shortOutput, "created", out, c.Out(), indent)
-	default:
-		afterFn = configcmd.NewPrintErrorAfter(mapper, c.Out())
-		afterFn = configcmd.HaltOnError(afterFn)
+
+	if !result.GeneratedJobs {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/newapp.go]: Leave the case statement

File: pkg/cmd/cli/cmd/newapp.go [hunk 2]
```diff
@@ -421,7 +412,16 @@ func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args
 	}
 	config.KubeClient = kclient
 	config.SetOpenShiftClient(osclient, namespace)
-	config.Out = out
+
+	// Only output="" and output="wide" should print descriptions of intermediate
+	// steps. Everything else, should print only some specific output (json, yaml,
+	// go-template, ...)
+	output := cmdutil.GetFlagString(c, "output")
+	if output == "" || output == "wide" {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/newapp.go]: Len(output) == 0

File: pkg/cmd/cli/cmd/newapp.go [hunk 3]
```diff
@@ -215,39 +211,38 @@ func RunNewApplication(fullName string, f *clientcmd.Factory, out io.Writer, c *
...
-	switch {
-	case shortOutput:
-		indent = ""
-	case len(output) != 0:
+	if output != "" && output != "wide" {
 		return f.Factory.PrintObject(c, result.List, out)
-	case !result.GeneratedJobs:
+	}
+
+	if !result.GeneratedJobs {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/newapp.go]: Use the switch statement

File: pkg/cmd/cli/cmd/newapp.go [hunk 4]
```diff
@@ -215,39 +211,38 @@ func RunNewApplication(fullName string, f *clientcmd.Factory, out io.Writer, c *
 		return err
 	}
 
-	indent := "    "
-	switch {
-	case shortOutput:
-		indent = ""
-	case len(output) != 0:
+	if output != "" && output != "wide" {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/newapp.go]: Always len(output) == 0

Comment (bparees) [inline pkg/cmd/cli/cmd/newapp.go]: or sometimes !=0, like in this case....

**Comment by bparees** [comment]
will need a squash.

**Comment by rhcarvalho** [comment]
Comments addressed, PTAL

**Comment by bparees** [comment]
what's the behavior with "-o wide" now?

**Comment by rhcarvalho** [comment]
> what's the behavior with "-o wide" now?

Not changing that in this PR.

**Comment by bparees** [comment]
so it still breaks?

**Comment by rhcarvalho** [comment]
'-o wide' is a separate matter we should fix independent of this (#6165).
It fails in different ways for new-app (where is it not a valid option) and new-build, I don't want to mix up with this PR. I prefer to keep it focused on fixing one thing.

**Comment by bparees** [comment]
That's fine, i just wanted to understand the situation.  thanks.

lgtm.

**Comment by bparees** [comment]
[merge]

**Comment by rhcarvalho** [comment]
pkg/generate/app/cmd unit tests broke, looking for a fix.

**Comment by rhcarvalho** [comment]
^^ broken tests were fixed here #6160

**Pull request merged**